client/interfaces.go

```diff
@@ -82,6 +91,12 @@ func (client *Client) Interfaces() (interfaces Interfaces, err error) {
 	return
 }
 
+// InterfaceInfos returns all interfaces, their meta-data, plugs and slots.
+func (client *Client) InterfaceInfos() (infos map[string]InterfaceInfo, err error) {
```

Comment:
This is sort of getting back to the prior design, except in a different endpoint. How about:

* Interface(name string) (*Interface, error)
* InterfaceNames() ([]string, error)

?